torch/csrc/autograd/engine.cpp

```diff
@@ -691,15 +691,14 @@ void validate_outputs(
       // AT_ERROR(format_error(ss.str()));
       continue;
     }
-    if (!grad.sizes().equals(metadata.shape())) {
-      if (!at::is_expandable_to(metadata.shape(), grad.sizes())) {
-        std::stringstream ss;
-        ss << "invalid gradient at index " << i << " - got ";
-        ss << grad.sizes() << " but expected shape compatible with ";
-        ss << metadata.shape();
-        AT_ERROR(format_error(ss.str()));
+
+    if (!metadata.is_same_shape(grad)) {
+      if (metadata.is_expandable_to_shape(grad)) {
+        grad = at::sum_to(std::move(grad), metadata.shape());
```

Comment:
[`sum_to`](https://github.com/pytorch/pytorch/blob/95b15c266baaf989ef7b6bbd7c23a2d90bacf687/aten/src/ATen/ExpandUtils.h#L440-L471) is also likely to need generalization eventually